### PR TITLE
Add Dockerized infrastructure

### PR DESCRIPTION
**Pull request by nelsonkopliku**
This PR adds a dockerized infrastructure providing:
- influxdb
- postgres

Adds a `help` to the makefile

image

image

### CODE REVIEW

**Review by dottorblaster — APPROVED** [review]
LGTM

File: Dockerfile [hunk 1]
```diff
@@ -3,7 +3,7 @@ ADD . /build
 WORKDIR /build
 RUN make build
 
-FROM gcr.io/distroless/base:debug AS trento-web
+FROM gcr.io/distroless/base:debug AS telemetry-collector
```
Comment (arbulu89) [inline Dockerfile]: Should be call it just `telemetry`? This is how the CI creates the container at least

File: Dockerfile [hunk 2]
```diff
@@ -3,7 +3,7 @@ ADD . /build
...
 COPY --from=go-build /build/telemetry /app/telemetry
```
Comment (fabriziosestito) [inline Dockerfile:7]: I think this was a leftover of my copy paste from the trento Dockerfile, we could also skip the `AS` directive or I would use `telemetry` as the target name

**Review by fabriziosestito — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -30,3 +30,20 @@ terraform init
...
+### Requirements
+
+1. [Docker](https://docs.docker.com/get-docker/)
+2. [Docker Compose](https://docs.docker.com/compose/install/)
+
+### Environment configuration
+
+Create a local environment file if needed: `cp .env.dist .env`
+
+Leverage odocker compose override feature by `cp docker-compose.override.yaml.dist docker-compose.override.yaml` for customization.
```
Comment (fabriziosestito) [inline README.md]: typo `odocker`

**Review by arbulu89 — COMMENTED** [review]
I would personally put the docker-compose elements in a `hack` folder as it is an auxiliary developer thing. Otherwise, I would include the telemetry container too as main deployment.

And now, as this is subjective, do as you wish hehe

File: README.md (hunk 3 above)
Comment (nelsonkopliku) [reply README.md]: fixed, thanks

File: Dockerfile (hunk 1 above)
Comment (nelsonkopliku) [reply Dockerfile]: changed

File: Dockerfile (hunk 2 above)
Comment (nelsonkopliku) [reply Dockerfile:7]: changed to `telemetry`

**Comment by nelsonkopliku** [comment]
> I would personally put the docker-compose elements in a `hack` folder as it is an auxiliary developer thing. Otherwise, I would include the telemetry container too as main deployment.
> 
> And now, as this is subjective, do as you wish hehe
>
We could, I don't have anything against or in favor.

That would mean adding a variable in the `Makefile`
```
DOCKER_COMPOSE_ROOT=./hack
```
and then explicitly provide docker-compose the correct paths everywhere docker-compose is involved.
The `start` target would change from
```
start:
        docker-compose up -d
```
to
```
start:
	docker-compose --project-directory=${DOCKER_COMPOSE_ROOT}  -f ${DOCKER_COMPOSE_ROOT}/docker-compose.yaml -f ${DOCKER_COMPOSE_ROOT}/docker-compose.override.yaml up -d
```

Which does the same, I find it a bit more noisy, though.

---

Yes, we could also define a custom function in `Makefile` like this
```
define docker_compose
	docker-compose --project-directory=${DOCKER_COMPOSE_ROOT}  -f ${DOCKER_COMPOSE_ROOT}/docker-compose.yaml -f ${DOCKER_COMPOSE_ROOT}/docker-compose.override.yaml $(1)
endef
```
and then call it like this
```
start:
	$(call docker_compose, up -d)
```
It works.

We know we can do it, yet I'would address this later, as you also propose eheh.

Thanks btw.

Edit:
@arbulu89 I didn't get this part:  _Otherwise, I would include the telemetry container too as main deployment._
Could you help me please? :sweat_smile:

**Pull request merged**